FSU-SP2023-SELab/group-1-lottopool
# Configure Frontend and Call Backend API

### Overview
Big pull request here, a lot of talk about.
- Added Tailwind CSS
- Removed boilerplate Solidjs code
- Added Solidjs router
  - Routes are contained in the routes folder
- Created demo where clicking a button will call the backend api endpoint /api/hello-world
  - This is part of the hello-world-example component and is called on the home-page route
  - Followed [Solidjs tutorials](https://www.solidjs.com/tutorial/async_resources) to ensure this was following best practices for async fetch calls (Solidjs is so much more enjoyable to use than React :))
  - Takes care of errors and loading appropriately, additionally refetches on each button press
- Adjusted some vscode config/formatter files
- Added flask_cors when running Flask in development (FLASK_DEBUG variable set to 1)
- Added additional frontend documentation
  - This was important to specify a strict file structure on the frontend to future-proof the file organization. I also want lots of code separation across files to ensure readability and reusability
  
 ### To test this PR
1. Open terminal for backend
- `cd backend`
- `pip install -r requirements.txt`
- `doppler run -- flask run`

2. Open terminal for frontend
- `cd frontend`
- `npm install`
- `doppler run -- npm start`

3. Clicking the button on the frontend should display a message. Additionally shutting the backend off will display an error message on button click.

Closes #1 
This PR should conclude Iteration 1
